testgrid/cmd/summarizer/summarizer.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"net/url"
+
+	responsepb "k8s.io/test-infra/testgrid/cmd/summarizer/response"
+	summarypb "k8s.io/test-infra/testgrid/summary"
+)
+
+// TableToSummary converts the input queryData collected from test results to
+// the DashboardTabSummary proto. It needs the dashboardName from the config
+// file and the updateTimestamp of when the updater collected the test results.
+func TableToSummary(queryData *responsepb.Response) (summarypb.DashboardTabSummary, error) {
+	s := extractBasicSummary(queryData)
+	s.FailingTestSummaries = extractTests(queryData)
+	return s, nil
+}
+
+func extractBasicSummary(queryData *responsepb.Response) summarypb.DashboardTabSummary {
+	bugURL := ""
+	if queryData.OpenBugTemplate != nil && len(queryData.OpenBugTemplate.Url) > 0 {
+		bugURL = queryData.OpenBugTemplate.Url
+	}
+	var lastRunTimestamp int64 = 0
+	for i, e := range queryData.Timestamps {
+		if i == 0 || e > lastRunTimestamp {
+			lastRunTimestamp = e
+		}
+	}
+	overallStatus := summarypb.DashboardTabSummary_UNKNOWN
```

Comment:
Can we just set overallStatus = queryData.OverallStatus?

Reply:
This is to provide the default value in case queryData.OverallStatus is empty. Otherwise, it will set to queryData.OverallStatus in the next two lines.

Reply:
Oh duh, the default for the enum is 0 not 1. Carry on!